pallets/pool-system/src/lib.rs

```diff
@@ -1239,15 +1240,39 @@ pub mod pallet {
 
 			let executed_amounts = epoch.tranches.fulfillment_cash_flows(solution)?;
 			let total_assets = epoch.nav.total(pool.reserve.total)?;
-			let tranche_ratios = epoch.tranches.combine_with_residual_top(
-				&executed_amounts,
-				|tranche, &(invest, redeem)| {
-					Ok(Perquintill::from_rational(
-						tranche.supply.ensure_add(invest)?.ensure_sub(redeem)?,
-						total_assets,
-					))
-				},
-			)?;
+
+			let tranche_ratios = {
+				let mut sum_non_residual_tranche_ratios = Perquintill::zero();
+				let num_tranches = pool.tranches.num_tranches();
+				let mut current_tranche = 1;
+				let mut ratios = epoch
+					.tranches
+					// NOTE: Reversing amounts, as residual amount is on top.
+					.combine_with_non_residual_top(
+						executed_amounts.rev(),
+						|tranche, &(invest, redeem)| {
+							let ratio = if current_tranche < num_tranches {
+								Perquintill::from_rational(
+									tranche.supply.ensure_add(invest)?.ensure_sub(redeem)?,
+									total_assets,
+								)
+							} else {
+								Perquintill::one().ensure_sub(sum_non_residual_tranche_ratios)?
```

Comment:
The sum of tranche ratios can never be higher than one, right?

Reply:
I think this line is annulated by line 1263. No matter which value `sum_non_residual_tranche_ratios` has, at the end of this iteration `sum_non_residual_tranche_ratios` will always have the value `one()`. Is this the expected behavior?